threedi_schema/application/threedi_database.py

```diff
@@ -35,43 +39,9 @@ def set_sqlite_pragma(dbapi_connection, connection_record):
     cursor.close()
 
 
-def load_spatialite(con, connection_record):
-    """Load spatialite extension as described in
-    https://geoalchemy-2.readthedocs.io/en/latest/spatialite_tutorial.html"""
-    import sqlite3
-
-    con.enable_load_extension(True)
-    cur = con.cursor()
-    libs = [
-        # SpatiaLite >= 4.2 and Sqlite >= 3.7.17, should work on all platforms
-        ("mod_spatialite", "sqlite3_modspatialite_init"),
-        # SpatiaLite >= 4.2 and Sqlite < 3.7.17 (Travis)
-        ("mod_spatialite.so", "sqlite3_modspatialite_init"),
-        # SpatiaLite < 4.2 (linux)
-        ("libspatialite.so", "sqlite3_extension_init"),
-    ]
-    found = False
-    for lib, entry_point in libs:
-        try:
-            cur.execute("select load_extension('{}', '{}')".format(lib, entry_point))
-        except sqlite3.OperationalError:
-            continue
-        else:
-            found = True
-            break
-    try:
-        cur.execute("select EnableGpkgAmphibiousMode()")
-    except sqlite3.OperationalError:
-        pass
-    if not found:
-        raise RuntimeError("Cannot find any suitable spatialite module")
-    cur.close()
-    con.enable_load_extension(False)
-
-
 class ThreediDatabase:
     def __init__(self, path, echo=False):
-        self.path = path
+        self._path = path
```

Comment:
This breaks the current working of `_path` and `path`. I introduced the setter to control the type of `ThreediDatabase.path` because it is set from multiple libraries, sometimes with a Path and sometimes with a string. With this implementation `self._path` is always a string and `self.path` will return a path; and setting `ThreediDatabase.path` will ensure that both `self._path` and `self.path` are set properly. 

It would be nicer to just have `self.path` and make is a `Path` (and add type hints). However, that also involves threedi-api; and I feel that this PR is complex enough as it is.

Note that `self._path` is now also cast to a string in the constructor 9fd2fe625343bfdd6836652d2529727d8f8cb889.

Reply:
Ok, I saw in the test that a Path instance was passed as parameter to ThreediDatabase(xxxx) # constructor
My suggestion also works if a string is passed. It's just converted into a Path.

The only thing that is not backwards-compatible is:
1) db = ThreediDatabase("/tmp/blaat.gpkg") and directly getting db.path as a string. (note: same new behaviour as your property introduced)
2) db = ThreediDatabase("/tmp/no_really_the.gpkg") ; db.path = "/tmp/the_real.gpkg". This is going to crash, but can be fixed by "path = self.path if isinstance(self.path, Path) else Path(self.path)" in the get_engine method.

Maybe 2 is actually the better solution. Convert it there to a Path instance if it not already the case:
```
path = self.path if isinstance(self.path, Path) else Path(self.path)
```
The property, setter and self._path can all be dropped and self.path stays exactly the same as what the user has put into the class constructor. And changing it after initialisation also works.